Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -231,6 +231,13 @@ static NonnullRefPtr<DynamicLoader> commit_elf(const String& name)
 void ELF::DynamicLinker::linker_main(String&& main_program_name, int main_program_fd, int argc, char** argv, char** envp)
 {
     g_envp = envp;
+    bool do_breakpoint_trap_before_entry = false;
+    for (char** env = envp; *env; ++env) {
+        if (StringView { *env } == "_LOADER_BREAKPOINT=1") {
+            do_breakpoint_trap_before_entry = true;
+        }
+    }
```

Comment:
I think this particular feature is fine, but we do have to be very careful when using the environment in the dynamic loader, since it will deal with user-provided input to set-uid programs.

Reply:
Agreed. Yeah, seems like the worst things a user could do in this case is make a set-uid program crash because of the uncaught SIGTRAP. The user won't be able to attach to the program because we disallow it ptrace.

Reply:
There's the AT_SECURE auxiliary value (which is set for setuid programs, I think?), which could be read to set a `s_allowed_to_check_env_vars` flag. I believe that's what other dynamic loaders do.

Reply:
That's neat, and it is indeed already set for setuid/gid programs.
I added a commit to this PR that does what you described here.